### PR TITLE
Interpolate universal tags #6821 

### LINKED ISSUE #6821: AWS: Add interpolation to the kms-key tag action
**Is your feature request related to a problem? Please describe.**
I'm not able to use '{now}' in the value when using the tag action for kms-key resources.

**Describe the solution you'd like**
I would like to be able to use '{now}' when tagging kms-key resources.

**Additional context**
I looked at the code a bit. The kms-key resource is using the `UniversalTag` class in `tags.py`, which unlike the `Tag` class in the same file that is used by many other AWS resource types, does not have an `interpolate_values` method. It's not a simple as just replicating that method in `UniversalTag` because the content of the `tags` variable is not in the same format in the two cases.

### PR DESCRIPTION
**Pull request by llange**
Implement interpolation for UniversalTag by reusing the implementation of ec2 tag interpolation.  
Tests are added - I was not able to properly use [`mock_datetime_now`](https://github.com/cloud-custodian/cloud-custodian/blob/50b1ee560dc833947176b1c094c9273317d9795b/c7n/testing.py#L272-L296) to freeze the time during the tests, so I'm using a [new dependency](https://github.com/spulec/freezegun) for this very need.
Do not hesitate to rewrite it with `mock_datetime_now` if you're luckier than I am.

Fixes #6821

### CODE REVIEW

**Comment by kapilt** [comment]
Thanks for the pull request!  If you don't mind could you sign the Linux Foundation/CNCF CLA referenced in the other comment. Direct link
https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/14202841/52837350/7532/#/?version=2

**Comment by llange** [comment]
Hi @kapilt , thanks for the comment !
Given that I already signed the previous forms of CLAs already existing in the past (E.g. https://github.com/cloud-custodian/cloud-custodian/pull/2089 ), and already contributed to the project previously (https://github.com/cloud-custodian/cloud-custodian/pulls?q=author%3Allange+is%3Amerged ) is there a way that you enable / migrate this information each time you change the CLA  / organization / .... ?
I do agree on the principle, but the process of sharing infos from my account GitHub (my email addresses - which are dedicated to GitHub) with an external entity (even as respected as The Linux Foundation) is not my preferred move at the moment.  
Any idea on how to clear the CLA requirements without sharing those infos ?
Thanks for your help.

**Comment by llange** [comment]
hi @kapilt , understood, and while I'm not a fan of this approach I did it as required (but I then revoked the Github access for the application).

**Comment by kapilt** [comment]
> hi @kapilt , understood, and while I'm not a fan of this approach I did it as required (but I then revoked the Github access for the application).

its appreciated, thank you

**Review by kapilt — APPROVED** [review]
thanks, lgtm

**Comment by llange** [comment]
Hi @kapilt, thanks for your approval.
Don't want to sound rude, but is there anything in the way of merging it to master ? Are you waiting for something from me that I forgot ?

**Comment by jnoble-rcd** [comment]
@kapilt Is there anyway we can get this merged? I was getting ready to work on this issue, until I saw this PR.

**Comment by nickcampau** [comment]
Is there anything remaining on this PR preventing it from being merged? This PR restores the functionality of existing policies that were broken when updated to a new version. This is a blocking issue for me and my team. If there is anything else needed we'd be  happy to assist.

File: c7n/tags.py [hunk 1]
```diff
@@ -876,6 +885,12 @@ def process_resource_set(self, client, resource_set, tags):
         return universal_retry(
             client.tag_resources, ResourceARNList=arns, Tags=tags)
 
+    def interpolate_values(self, tags):
+        """Interpolate in a list of tags - 'new' resourcegroupstaggingapi format
+        """
+        for key in list(tags.keys()):
```
Comment (ajkerrigan) [inline c7n/tags.py:891]: nit - `for key in tags` will work just as well here

**Review by ajkerrigan — APPROVED** [review]
looked like there was a merge conflict resolved and we're up to date now. this was approved previously and looks good from here :+1:. should be good to merge once the checks come back green. thanks for sticking with this one!

**Pull request merged**